webui/react/src/contexts/Auth.tsx

```diff
@@ -1,22 +1,33 @@
-import { CancelToken } from 'axios';
-import { Dispatch } from 'react';
-
 import { generateContext } from 'contexts';
-import { getCurrentUser, isAuthFailure } from 'services/api';
 import { Auth } from 'types';
 
 enum ActionType {
   Reset,
+  ResetCheck,
   Set,
+  UpdateCheck,
 }
 
-type State = Auth;
+/*
+ * `checkCount` allows the `useAuthCheck` hook to keep tabs of how many times
+ * is has been called in sign in. It is kept here to avoid a situation where
+ * `isAuthenticated` is off sync with `checkCount`, which causes the Sign In
+ * form to flicker briefly before being redirected to an authenticated page.
+ */
+type State = Auth & {
+  checked: boolean;
+};
 
 type Action =
-  | { type: ActionType.Reset}
+  | { type: ActionType.Reset }
+  | { type: ActionType.ResetCheck }
   | { type: ActionType.Set; value: Auth }
+  | { type: ActionType.UpdateCheck }
```

Comment:
nit: name it something that conveys this always sets checked=true .. maybe `markChecked`?